### PR TITLE
[Subtitles] Fix wrong language from subtitle name

### PR DESCRIPTION
**Pull request by matthuisman**
## Description

change the subtitle logic to parse the subtitle name backwards when looking for language & flags.
https://forum.kodi.tv/showthread.php?tid=365802

## Motivation and context

when setting a subtitle programmatically, kodi doesnt know the movie name so cant remove the movie name from the subtitle name. This leads to the in-correct language being obtained from the movie name. 

Code here to confirm the issue
https://forum.kodi.tv/showthread.php?tid=365802&pid=3074128#pid3074128

## How has this been tested?

user has compiled and tested:
https://forum.kodi.tv/showthread.php?tid=365802&pid=3075199#pid3075199

"Back to the Future Part III (1990) [tt0099088] English.srt"
CURRENT:
Language = 'ton' / Name = 'Back the Future Part III (1990) [tt0099088] English (External)'
WITH PR:
Language = 'eng' / Name = 'Back to the Future Part III (1990) [tt0099088] (External)' 

"No Time to Die (2021) English Forced.srt"
CURRENT:
Language = 'nor' / Name = 'Time to Die (2021) English (External)'
WITH PR:
 Language = 'eng' / Name = 'No Time to Die (2021) (External)'

## What is the effect on users?

Correct language and flags obtained and correct filename

## Types of change

- [X] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/Util.cpp [hunk 1]
```diff
@@ -2114,20 +2114,8 @@ ExternalStreamInfo CUtil::GetExternalStreamDetailsFromFilename(const std::string
     std::vector<std::string> tokens;
     StringUtils::Tokenize(inputString, tokens, delimiters);
 
-    for (std::vector<std::string>::iterator it = tokens.begin(); it != tokens.end(); ++it)
+    for (std::vector<std::string>::reverse_iterator it = tokens.rbegin(); it != tokens.rend(); ++it)
```
Comment (CastagnaIT) [inline xbmc/Util.cpp:2117]: `std::vector<std::string>::reverse_iterator` iterator can be changed with `auto`

File: xbmc/Util.cpp [hunk 2]
```diff
@@ -2147,7 +2135,19 @@ ExternalStreamInfo CUtil::GetExternalStreamDetailsFromFilename(const std::string
         continue;
       }
 
-      name += " " + (*it);
+      if (info.language.empty())
+      {
+        std::string langTmp(*it);
+        std::string langCode;
+        // try to recognize language
+        if (g_LangCodeExpander.ConvertToISO6392B(langTmp, langCode))
```
Comment (CastagnaIT) [inline xbmc/Util.cpp:2143]: define the variable `langTmp` seem not needed,
can you remove it? and use `*it`

**Comment by CastagnaIT** [comment]
Please add `[Subtitles]` prefix on your commit title

File: xbmc/Util.cpp (hunk 1 above)
Comment (matthuisman) [reply xbmc/Util.cpp]: done - i think... :)

File: xbmc/Util.cpp (hunk 2 above)
Comment (matthuisman) [reply xbmc/Util.cpp]: good catch. done

**Review by enen92 — APPROVED** [review]

**Review by AlwinEsch — APPROVED** [review]

**Comment by matthuisman** [comment]
wow, that was a quick merge :)
Should I do a backport PR for Matrix?

**Comment by matthuisman** [comment]
so was trying this with latest nightly and thought it had broken external subs.
But tried the nightly before it and had the same issue.

Seems setSubtitles doesn't work in kodi 20 with local files...
eg. li.setSubtitles(['D:\\Back to the Future Part III (1990).eng.srt']) works fine on Kodi 19, but not on 20

**Pull request merged**